autoload/neomake/utils.vim

```diff
@@ -379,16 +379,15 @@ function! neomake#utils#ExpandArgs(args) abort
     let isk = &iskeyword
     set iskeyword=p,h,t,r,e,%,:
     try
-        let home = expand('~')
         let ret = map(a:args,
                     \ 'substitute(v:val, '
-                    \ . '''\(\%(\\\@<!\\\)\@<!\~\)'', '
-                    \ . 'home, '
+                    \ . '''\(\%(\\\@<!\\\)\@<!%\%(%\|\%(:[phtre]\+\)*\)\ze\)\w\@!'', '
+                    \ . '''\=(submatch(1) == "%%" ? "%" : expand(submatch(1)))'', '
                     \ . '''g'')')
         let ret = map(ret,
                     \ 'substitute(v:val, '
-                    \ . '''\(\%(\\\@<!\\\)\@<!%\%(%\|\%(:[phtre]\+\)*\)\ze\)\w\@!'', '
-                    \ . '''\=(submatch(1) == "%%" ? "%" : expand(submatch(1)))'', '
+                    \ . '''\(\%(\\\@<!\\\)\@<!\~\)'', '
+                    \ . 'expand(''~''), '
```

Comment:
I thought it would be better, in case $HOME expands to something that would get picked up later then.